### DISCUSSION TITLE
How can I have json.dump have similar to std::cout way for printing "//"?

### DISCUSSION THREAD

**Discussion by sreerajchundayil** [nlohmann/json#3292] [Q&A] [answered]
When we do `cout` of the `str` below we get one `\` in terminal.
But dumping into json is not doing it. I don't have the control of the input string.
Is there any parameter which I can configure to do it.

Current output in json:
"\\\\"
Expected output in json:
"\\"

```
#include <fstream>
#include <iostream>
#include <string>
#include "json.h"

using namespace std;
using json = nlohmann::json;


int main()
{
  std::string str = "\\"; 
  nlohmann::json js = str;
  ofstream ofs("a");
  ofs << js.dump(2);
}
```

**Comment by nlohmann** [comment, marked as answer]
I cannot see any differences in the created file and `js.dump(2)` streamed to `std::cout`. It is `"\\"` in both cases.

JSON requires backslashes to be properly escaped. `"\"` would be invalid JSON.

**Comment by sreerajchundayil** [comment]
```
std::string str = "\\"; 
std::cout << str << std::endl; // gives just \
std::cout << js.dump(2) << std::endl; // gives "\\"
```

**Reply by nlohmann** [reply]
Yes, this is because the second link prints the string `str` and the third one the JSON serialization of it.

**Reply by nlohmann** [reply]
(See https://datatracker.ietf.org/doc/html/rfc8259#section-7)

**Reply by nlohmann** [reply]
@sreerajchundayil Do you need further assistance with this issue?

**Comment by sreerajchundayil** [comment]
@nlohmann : No. I am ok with the answer.
